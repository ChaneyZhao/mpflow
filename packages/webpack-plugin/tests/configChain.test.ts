import WebpackPlugin from '@mpflow/webpack-plugin'

const { ConfigChain } = WebpackPlugin

describe('ConfigChain', () => {
  test('should work', () => {
    const config = new ConfigChain()

    config.resolve.roots.add('/virtualRoot')
    config.resolve.javascript.extensions.add('.ts')
    config.resolve.wxss.extensions.add('.css').add('.less').add('.sass').add('.scss').add('.stylus')

    config.rules
      .wxss('less')
      .test(/\.less$/)
      .use('less-loader')
      .loader('less-loader')
    config.rules
      .wxss('sass')
      .test(/\.sass$/)
      .use('sass-loader')
      .loader('sass-loader')
      .options({ indentedSyntax: true })
    config.rules
      .wxss('scss')
      .test(/\.scss$/)
      .use('sass-loader')
      .loader('sass-loader')
    config.rules
      .wxss('stylus')
      .test(/\.stylus$/)
      .use('stylus-loader')
      .loader('stylus-loader')

    config.program.appId('wx123')
    config.program.compileType('miniprogram')
    config.program.projectName('test')
    config.program.miniprogramRoot('/miniprogram')
    config.program.pluginRoot('/plugin')
    config.program.qcloudRoot('/cloud')

    expect(config.toConfig()).toMatchSnapshot('toConfig')
    expect(config.toString()).toMatchSnapshot('toString')
  })
})
