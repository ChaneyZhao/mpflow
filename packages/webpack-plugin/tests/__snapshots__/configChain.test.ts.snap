// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigChain should work: toConfig 1`] = `
Object {
  "externals": Array [
    /\\^weui-miniprogram\\\\//,
  ],
  "program": Object {
    "appId": "wx123",
    "compileType": "miniprogram",
    "miniprogramRoot": "/miniprogram",
    "pluginRoot": "/plugin",
    "projectName": "test",
    "qcloudRoot": "/cloud",
    "settings": Object {
      "es6": false,
    },
  },
  "resolve": Object {
    "javascript": Object {
      "extensions": Array [
        ".ts",
      ],
    },
    "roots": Array [
      "/virtualRoot",
    ],
    "wxss": Object {
      "extensions": Array [
        ".css",
        ".less",
        ".sass",
        ".scss",
        ".stylus",
      ],
    },
  },
  "rules": Object {
    "wxss": Array [
      Object {
        "test": /\\\\\\.less\\$/,
        "use": Array [
          Object {
            "loader": "less-loader",
          },
        ],
      },
      Object {
        "test": /\\\\\\.sass\\$/,
        "use": Array [
          Object {
            "loader": "sass-loader",
            "options": Object {
              "indentedSyntax": true,
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.scss\\$/,
        "use": Array [
          Object {
            "loader": "sass-loader",
          },
        ],
      },
      Object {
        "test": /\\\\\\.stylus\\$/,
        "use": Array [
          Object {
            "loader": "stylus-loader",
          },
        ],
      },
    ],
  },
}
`;

exports[`ConfigChain should work: toString 1`] = `
"{
  resolve: {
    roots: [
      '/virtualRoot'
    ],
    javascript: {
      extensions: [
        '.ts'
      ]
    },
    wxss: {
      extensions: [
        '.css',
        '.less',
        '.sass',
        '.scss',
        '.stylus'
      ]
    }
  },
  rules: {
    wxss: [
      /* config.module.rule('less') */
      {
        test: /\\\\.less$/,
        use: [
          /* config.module.rule('less').use('less-loader') */
          {
            loader: 'less-loader'
          }
        ]
      },
      /* config.module.rule('sass') */
      {
        test: /\\\\.sass$/,
        use: [
          /* config.module.rule('sass').use('sass-loader') */
          {
            loader: 'sass-loader',
            options: {
              indentedSyntax: true
            }
          }
        ]
      },
      /* config.module.rule('scss') */
      {
        test: /\\\\.scss$/,
        use: [
          /* config.module.rule('scss').use('sass-loader') */
          {
            loader: 'sass-loader'
          }
        ]
      },
      /* config.module.rule('stylus') */
      {
        test: /\\\\.stylus$/,
        use: [
          /* config.module.rule('stylus').use('stylus-loader') */
          {
            loader: 'stylus-loader'
          }
        ]
      }
    ]
  },
  externals: [
    /^weui-miniprogram\\\\//
  ],
  program: {
    appId: 'wx123',
    compileType: 'miniprogram',
    projectName: 'test',
    miniprogramRoot: '/miniprogram',
    pluginRoot: '/plugin',
    qcloudRoot: '/cloud',
    settings: {
      es6: false
    }
  }
}"
`;
